cmd/restic/cmd_snapshots.go

```diff
@@ -45,6 +47,13 @@ func init() {
 	f.StringArrayVar(&snapshotOptions.Paths, "path", nil, "only consider snapshots for this `path` (can be specified multiple times)")
 	f.BoolVarP(&snapshotOptions.Compact, "compact", "c", false, "use compact format")
 	f.BoolVar(&snapshotOptions.Last, "last", false, "only show the last snapshot for each host and path")
+	f.StringVarP(&snapshotOptions.GroupBy, "group-by", "g", "", "string for grouping snapshots by host,paths,tags")
+}
+
+type groupKey struct {
+	Hostname string
```

Comment:
Add so-called struct tags here to set the name for the JSON output (lowercase, snake_case)
```suggestion
	Hostname string `json:"hostname"`
```